internal/scheduler/compaction.go

```diff
@@ -0,0 +1,126 @@
+package scheduler
```

Comment:
Is this file in a usable state? The commented out code means that the structs don't comform to the interface and the todos mean that the implementations are not thread safe.